### PR TITLE
AP_GPS: fix time_epoch_usec

### PR DESCRIPTION
**Pull request by bugobliterator**
fixes issues mentioned here https://github.com/ArduPilot/ardupilot/issues/19963

### CODE REVIEW

**Review by hendjoshsr71 — APPROVED** [review]

File: libraries/AP_GPS/GPS_Backend.h [hunk 1]
```diff
@@ -95,7 +95,7 @@ class AP_GPS_Backend
 
     // return iTOW of last message, or zero if not supported
     uint32_t get_last_itow(void) const {
-        return (_pseudo_itow_delta_ms == 0)?(_last_itow*1000ULL):((_pseudo_itow/1000ULL) + _pseudo_itow_delta_ms);
+        return (_pseudo_itow_delta_ms == 0)?(_last_itow):((_pseudo_itow/1000ULL) + _pseudo_itow_delta_ms);
```
Comment (peterbarker) [inline libraries/AP_GPS/GPS_Backend.h:98]: Do you think we should rename `_last_itow` to `_last_itow_ms`?  That may have stopped this bug creeping in.

File: libraries/AP_GPS/AP_GPS.cpp [hunk 2]
```diff
@@ -522,7 +522,7 @@ uint64_t AP_GPS::time_epoch_usec(uint8_t instance) const
         fix_time_ms = time_epoch_convert(istate.time_week, drivers[instance]->get_last_itow());
         return (fix_time_ms*1000ULL) + (AP_HAL::micros64() - istate.last_corrected_gps_time_us);
     } else {
-        fix_time_ms = time_epoch_convert(istate.time_week, istate.time_week_ms) * 1000ULL;
+        fix_time_ms = time_epoch_convert(istate.time_week, istate.time_week_ms);
```
Comment (peterbarker) [inline libraries/AP_GPS/AP_GPS.cpp:525]: This is a pretty uninformative name for a funciton, really.  We should probably covert it...
`time_epoch_convert` -> `istate_time_to_epoch_ms`?

File: libraries/AP_GPS/AP_GPS_MAV.cpp [hunk 3]
```diff
@@ -123,6 +123,9 @@ void AP_GPS_MAV::handle_msg(const mavlink_message_t &msg)
                 uint32_t corrected_ms = jitter.correct_offboard_timestamp_msec(timestamp_ms, now_ms);
                 state.last_corrected_gps_time_us = (corrected_ms * 1000ULL);
                 state.corrected_timestamp_updated = true;
+                if (state.last_corrected_gps_time_us) {
+                    _last_itow = state.time_week_ms;
+                }
```
Comment (peterbarker) [inline libraries/AP_GPS/AP_GPS_MAV.cpp:128]: Could we get a comment in here as to why `last_corrected_gps_time_us` is a reasonable gate to us setting from `state.time_week_ms`?

**Comment by shellixyz** [comment]
Tested, seems to be working like it should now, thank you. Closes #19963

**Pull request merged**